From 7349c1125423555d908d89a0fd1548cf5cee2de3 Mon Sep 17 00:00:00 2001
From: jocelyn.turcotte <jocelyn.turcotte@digia.com>
Date: Mon, 15 Apr 2013 13:52:35 +0200
Subject: [PATCH] [Qt] REGRESSION(r137436): It made all inspector tests timeout on developer builds
 https://bugs.webkit.org/show_bug.cgi?id=106554

Reviewed by Simon Hausmann.

.:

Explicitely link WebCore resources in the final DLL only on Windows to
support force_static_libs_as_shared on other platforms.

WebKit1 applications don't get the QtWebKit dynamic library loaded
since libQtWebKitWidgets doesn't depend on libQtWebKit if WebCore and
WebKit1 are dynamic libraries of their own.

* Source/api.pri:

Source/WebCore:

* Target.pri:

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@141892 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 ChangeLog                 |   16 ++++++++++++++++
 Source/WebCore/ChangeLog  |   33 +++++++++++++++++++++------------
 Source/WebCore/Target.pri |   17 ++++++++++-------
 Source/api.pri            |   13 +++++++++++++
 4 files changed, 60 insertions(+), 19 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 2bcd7e9..3943f4d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,19 @@
+2013-02-05  Jocelyn Turcotte  <jocelyn.turcotte@digia.com>
+
+        [Qt] REGRESSION(r137436): It made all inspector tests timeout on developer builds
+        https://bugs.webkit.org/show_bug.cgi?id=106554
+
+        Reviewed by Simon Hausmann.
+
+        Explicitely link WebCore resources in the final DLL only on Windows to
+        support force_static_libs_as_shared on other platforms.
+
+        WebKit1 applications don't get the QtWebKit dynamic library loaded
+        since libQtWebKitWidgets doesn't depend on libQtWebKit if WebCore and
+        WebKit1 are dynamic libraries of their own.
+
+        * Source/api.pri:
+
 2012-11-30  Simon Hausmann  <simon.hausmann@digia.com>
 
         [Qt] Unreviewed doc fix
diff --git a/Source/WebCore/ChangeLog b/Source/WebCore/ChangeLog
index 8bb2fd3..1184440 100644
--- a/Source/WebCore/ChangeLog
+++ b/Source/WebCore/ChangeLog
@@ -275,18 +275,6 @@
         (audio::-webkit-media-controls-return-to-realtime-button, video::-webkit-media-controls-return-to-realtime-button):
         (audio::-webkit-media-controls-toggle-closed-captions-button, video::-webkit-media-controls-toggle-closed-captions-button):
 
-2012-09-12  Simon Hausmann  <simon.hausmann@nokia.com>
-
-        [Qt] Drastically shorten length of commandline needed for JS bindings generator
-        https://bugs.webkit.org/show_bug.cgi?id=96266
-
-        Reviewed by Tor Arne Vestbø.
-
-        The generate-bindings script supports the SOURCE_ROOT environment variable for IDL include file
-        lookups, which allows specifying relative include search directories.
-
-        * DerivedSources.pri:
-
 2013-02-12  Robert Hogan  <robert@webkit.org>
 
         REGRESSION(r136967): Combination of float and clear yields to bad layout
@@ -314,6 +302,15 @@
         * rendering/RenderFrameSet.cpp:
         (WebCore::RenderFrameSet::paint):
 
+2013-02-05  Jocelyn Turcotte  <jocelyn.turcotte@digia.com>
+
+        [Qt] REGRESSION(r137436): It made all inspector tests timeout on developer builds
+        https://bugs.webkit.org/show_bug.cgi?id=106554
+
+        Reviewed by Simon Hausmann.
+
+        * Target.pri:
+
 2013-02-01  Philip Rogers  <pdr@google.com>
 
         Prevent skipped repaints for children of inner SVG elements
@@ -70818,4 +70815,16 @@ ayer
         * platform/graphics/WOFFFileFormat.cpp:
         * platform/graphics/chromium/VDMXParser.cpp:
 
+2012-09-12  Simon Hausmann  <simon.hausmann@nokia.com>
+
+        [Qt] Drastically shorten length of commandline needed for JS bindings generator
+        https://bugs.webkit.org/show_bug.cgi?id=96266
+
+        Reviewed by Tor Arne Vestbø.
+
+        The generate-bindings script supports the SOURCE_ROOT environment variable for IDL include file
+        lookups, which allows specifying relative include search directories.
+
+        * DerivedSources.pri:
+
 == Rolled over to ChangeLog-2012-10-19 ==
diff --git a/Source/WebCore/Target.pri b/Source/WebCore/Target.pri
index b3435a6..98eea8c 100644
--- a/Source/WebCore/Target.pri
+++ b/Source/WebCore/Target.pri
@@ -20,13 +20,16 @@ DEFINES += QT_MAKEDLL
     INCLUDEPATH += $$PWD/../WTF/wtf/qt/compat
 }
 
-RESOURCES += \
-    $$PWD/WebCore.qrc
-
-include_webinspector {
-    RESOURCES += \
-        $$PWD/inspector/front-end/WebKit.qrc \
-        $${WEBCORE_GENERATED_SOURCES_DIR}/InspectorBackendCommands.qrc
+# Do it in the WebCore static lib to support force_static_libs_as_shared
+# since the QtWebKitWidgets lib wouldn't load QtWebKit in that case.
+# This should match the opposite statement in api.pri for the QtWebKit lib.
+!win* {
+    RESOURCES += $$PWD/WebCore.qrc
+    include_webinspector {
+        RESOURCES += \
+            $$PWD/inspector/front-end/WebKit.qrc \
+            $${WEBCORE_GENERATED_SOURCES_DIR}/InspectorBackendCommands.qrc
+    }
 }
 
 SOURCES += \
diff --git a/Source/api.pri b/Source/api.pri
index 9da4b9f..2c30283 100644
--- a/Source/api.pri
+++ b/Source/api.pri
@@ -99,6 +99,19 @@ QMAKE_DOCS = $$PWD/qtwebkit.qdocconf
 !no_webkit1: WEBKIT += webkit1
 !no_webkit2: WEBKIT += webkit2
 
+# Resources have to be included directly in the final binary for MSVC.
+# The linker won't pick them from a static library since they aren't referenced.
+win* {
+    RESOURCES += $$PWD/WebCore/WebCore.qrc
+    include_webinspector {
+        # WEBCORE_GENERATED_SOURCES_DIR is defined in WebCore.pri, included by
+        # load(webkit_modules) if WEBKIT contains webcore.
+        RESOURCES += \
+            $$PWD/WebCore/inspector/front-end/WebKit.qrc \
+            $${WEBCORE_GENERATED_SOURCES_DIR}/InspectorBackendCommands.qrc
+    }
+}
+
 # ------------- Install rules -------------
 
 haveQt(5) {
-- 
1.7.1

